language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -727,18 +728,27 @@ instance ToJSON PackageJson where
 
 -- Read the provided 'package.json'; transform it to include the
 -- provided workspaces; write it back to disk.
-setupWorkspace :: FilePath -> [(T.Text, [Dependency])] -> FilePath -> IO ()
-setupWorkspace optOutputDir dependencies file = do
+setupWorkspace :: FilePath -> FilePath -> [(T.Text, [Dependency])] -> IO ()
+setupWorkspace optInputPackageJson optOutputDir dependencies = do
+  -- If the file designated by 'optInputPackageJson' doesn't yet
+  -- exist, create it.
+  packageJsonExists <- doesFileExist optInputPackageJson
+  when (not packageJsonExists) $ do
```

Comment:
Maybe catch the error when reading the file and default it to this instead of explicitly creating it here?